### DISCUSSION TITLE
Dynamic SQL with an array variable in a DuckDB 'IN'?

### DISCUSSION THREAD

**Discussion by mskullcap** [observablehq/framework#1088] [Q&A] [answered]
I am struggling to generate dynamic sql for array values.  The SQL I want to generate should look like this:

```sql
SELECT * FROM events
WHERE name IN ('Sputnik 1', 'Apollo 11')
```

and if I put this in a sql template, it works.  The challenge I have is if I define the values as an array and then try to run the SQL generated from the array:

```js
const sc = ["Sputnik 1", "Apollo 11"]
```

```sql
SELECT * FROM events
WHERE name IN (${sc.join(",")})
```

I just can't get this sql to work.  If I put `${sc[0]}`, it works for the one element.  Nothing I do works with the array; if I display() the generated SQL, the string looks fine, 

[events.md](https://github.com/observablehq/framework/files/14627106/events.md)

**Comment by Fil** [comment, marked as answer]
this is not yet supported (by DuckDB) https://github.com/duckdb/duckdb-wasm/issues/447

the way around it is to do your own escaping; something like:

```js
sql([
  `SELECT x, y FROM t WHERE x IN (${
    selection.length
      ? selection.map((d) => `'${d.replaceAll("'", "''")}'`)
      : "NULL"
  })`
])
```

**Reply by mskullcap** [reply]
Thank you @Fil ... I missed that.
